Unit Tests/Tests/Chatkit/ChatkitTests.swift

```diff
@@ -1,23 +1,130 @@
 import XCTest
+import Mockingjay
 import PusherPlatform
 @testable import PusherChatkit
 
 class ChatkitTests: XCTestCase {
     
+    // MARK: - Tests lifecycle
+    
+    override func tearDown() {
+        removeAllStubs()
+        
+        super.tearDown()
+    }
+    
     // MARK: - Tests
     
     func testShouldSetRequiredConfiguration() {
-        let chatkit = try? Chatkit(instanceLocator: "testInstanceLocator", tokenProvider: TestTokenProvider(), logger: PPDefaultLogger())
+        let chatkit = try? Chatkit(instanceLocator: Networking.testInstanceLocator, tokenProvider: TestTokenProvider(), logger: TestLogger())
         
-        XCTAssertEqual(chatkit?.instanceLocator, "testInstanceLocator")
+        XCTAssertEqual(chatkit?.instanceLocator, Networking.testInstanceLocator)
         XCTAssertTrue(chatkit?.tokenProvider is TestTokenProvider)
-        XCTAssertTrue(chatkit?.logger is PPDefaultLogger)
+        XCTAssertTrue(chatkit?.logger is TestLogger)
+    }
+    
+    func testShouldHaveDefaultLogger() {
+        let chatkit = try? Chatkit(instanceLocator: Networking.testInstanceLocator, tokenProvider: TestTokenProvider())
+        
+        XCTAssertNotNil(chatkit?.logger)
+    }
+    
+    func testShouldNotInstantiateWithInstanceLocatorInIncorrectFormat() {
+        let chatkit = try? Chatkit(instanceLocator: "invalidInstanceLocator", tokenProvider: TestTokenProvider())
+        
+        XCTAssertNil(chatkit)
+    }
+    
+    func testShouldHaveConnectionStatusSetToDisconnectedAfterInstantiation() {
+        let chatkit = try? Chatkit(instanceLocator: Networking.testInstanceLocator, tokenProvider: TestTokenProvider())
+        
+        XCTAssertEqual(chatkit?.connectionStatus, ConnectionStatus.disconnected)
+    }
+    
+    func testShouldReturnNoErrorAfterSuccessfulConnection() {
+        stubSubscription(of: .chat, version: .version6, instanceLocator: Networking.testInstanceLocator, path: .users, with: "chat-initial_state")
+        
+        let chatkit = try? Chatkit(instanceLocator: Networking.testInstanceLocator, tokenProvider: TestTokenProvider())
+        
+        let expectation = self.expectation(description: "Connection")
+        
+        chatkit?.connect { error in
+            XCTAssertNil(error)
```

Comment:
Would it be beneficial if we would check also the HTTP response code and error message?

Reply:
Platform SDK does not provide us with such information.